### PR TITLE
Set WorkingDirectory for terminal

### PR DESCRIPTION
**Pull request by hez2010**
- Set WorkingDirectory for terminal so that cmd no longer need to execute `cd` command for switching directory.

### CODE REVIEW

**Comment by lampenlampen** [comment]
Do you know, if this works for the other Terminal apps as well?

**Comment by hez2010** [comment]
> Do you know, if this works for the other Terminal apps as well?

Tested with cmd, powershell and wsl. 
Major shells will set WorkingDirectory as current directory such as cmd, powershell, bash and etc.

Windows Terminal needs explicit argument "-d {0}".

File: Files/Views/SettingsPages/Appearance.xaml.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public Appearance()
             ThemeChooser.SelectedIndex = (int)Enum.Parse(typeof(ElementTheme), ThemeHelper.RootTheme.ToString());
             ThemeChooser.Loaded += (s, e) =>
             {
-                ThemeChooser.SelectionChanged += async (s1, e1) =>
```
Comment (tsvietOK) [inline Files/Views/SettingsPages/Appearance.xaml.cs:28]: What purpose of this change?

Comment (hez2010) [reply Files/Views/SettingsPages/Appearance.xaml.cs:28]: Without `await` in the closure, `async` here is unnecessary but adds overhead. VS also produces warnings for it.

Comment (tsvietOK) [reply Files/Views/SettingsPages/Appearance.xaml.cs:28]: Did you test this change? Is changing theme working properly?

Comment (hez2010) [reply Files/Views/SettingsPages/Appearance.xaml.cs:28]: Sorry for the out-of-scope change, but I've tested it and it works fine.
image
image

File: Files/Assets/terminal/terminal.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     {
       "name": "CMD",
       "path": "cmd.exe",
-      "arguments": "/k \"cd /d {0} && title Command Prompt\"",
```
Comment (tsvietOK) [inline Files/Assets/terminal/terminal.json:8]: This change will not update user's `terminal.json`. So will it work with old arguments?

Comment (hez2010) [reply Files/Assets/terminal/terminal.json:8]: Yes, old arguments still work.

File: Files/Interacts/Interaction.cs [hunk 3]
```diff
@@ -138,9 +138,12 @@ public async void OpenDirectoryInTerminal(object sender, RoutedEventArgs e)
...
             if (App.Connection != null)
             {
-                var value = new ValueSet();
-                value.Add("Application", terminal.Path);
-                value.Add("Arguments", string.Format(terminal.Arguments, CurrentInstance.ViewModel.WorkingDirectory));
+                var value = new ValueSet
+                {
+                    { "WorkingDirectory", CurrentInstance.ViewModel.WorkingDirectory },
+                    { "Application", terminal.Path },
+                    { "Arguments", string.Format(terminal.Arguments, CurrentInstance.ViewModel.WorkingDirectory) }
```
Comment (tsvietOK) [inline Files/Interacts/Interaction.cs:145]: To fix Windows Terminal opening after app launch:
```suggestion
                    { "Arguments", string.Format(terminal.Arguments, InstanceTabsView.NormalizePath(CurrentInstance.ViewModel.WorkingDirectory)) }
```

Comment (hez2010) [reply Files/Interacts/Interaction.cs]: Done.
Also change default arguments for other terminals and tested ok.

File: Files/View Models/SelectedItemsPropertiesViewModel.cs [hunk 4]
```diff
@@ -150,6 +150,13 @@ public string ItemMD5Hash
                 }
             }
         }
+        private bool _ItemMD5HashCalcError;
```
Comment (tsvietOK) [inline Files/View Models/SelectedItemsPropertiesViewModel.cs:153]: Remove it from this PR.

File: Files/Assets/terminal/terminal.json [hunk 5]
```diff
@@ -12,13 +12,13 @@
     {
       "name": "PowerShell Core 6",
```
[the remark this answers is not in the document]
Comment (tsvietOK) [reply Files/Assets/terminal/terminal.json:13]: It is not needed, this profile here is just for example.

File: Files/View Models/SelectedItemsPropertiesViewModel.cs (hunk 4 above)
Comment (hez2010) [reply Files/View Models/SelectedItemsPropertiesViewModel.cs:153]: Sorry but I don't know why my changes on another branch be commit to this branch....

Comment (hez2010) [reply Files/View Models/SelectedItemsPropertiesViewModel.cs]: Done

File: Files/Assets/terminal/terminal.json [hunk 6]
```diff
@@ -5,20 +5,20 @@
     {
       "name": "CMD",
       "path": "cmd.exe",
-      "arguments": "/k \"cd /d {0} && title Command Prompt\"",
+      "arguments": "",
       "icon": ""
     }
     /*
```
Comment (yaichenbaum) [inline Files/Assets/terminal/terminal.json:11]: By removing the comment, what would happen if Powershell is not installed?

Comment (duke7553) [reply Files/Assets/terminal/terminal.json:11]: It is an example profile which users can derive new profiles from.

Comment (yaichenbaum) [reply Files/Assets/terminal/terminal.json:11]: Correct, but by removing the `*/` it adds it is actually adding it as one of the profiles.

Comment (lukeblevins) [reply Files/Assets/terminal/terminal.json:11]: @yair100 I just viewed the raw file, and they don't appear to be removed

Comment (duke7553) [reply Files/Assets/terminal/terminal.json:11]: image

**Pull request merged**